<?xml version="1.0" encoding="utf-8"?>
<launch>

  <!-- put file path of urdf model in parameter server, for the cylinder spawner node -->
  <!-- more details at launch file of swarm robot simulation -->
  <param name="red_cylinder_path" type="str" value="/home/grady/test_ws/src/two_scara_collaboration/urdf/red_cylinder.urdf"/>
  <param name="blue_cylinder_path" type="str" value="/home/grady/test_ws/src/two_scara_collaboration/urdf/blue_cylinder.urdf"/>

  <!-- initialize gazebo with a world file -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find two_scara_collaboration)/launch/initialize.world"/>
    <arg name="gui" value="true" />
    <arg name="paused" value="true"/>
    <arg name="use_sim_time" value="true"/>
  </include>
  <!-- gazebo has to be paused when starts up -->
  <!-- because roslaunch doesn't guarantee serial running of each node -->
  <!-- in this case, two scara will hit each other and swing back and forth -->


  <!-- spawn the conveyor_belt in gazebo -->
  <node name="spawn_conveyor_belt" pkg="gazebo_ros" type="spawn_model" args="-file $(find two_scara_collaboration)/urdf/conveyor_belt.urdf -urdf -model conveyor_belt" />

  <!-- spawn the right scara robot in gazebo -->
  <node name="spawn_scara_robot_right" pkg="gazebo_ros" type="spawn_model" args="-file $(find two_scara_collaboration)/urdf/scara_robot_right.urdf -urdf -model scara_robot_right
    -y -1.5
    -J scara_robot_right::rotation1 -0.78
    -J scara_robot_right::rotation2 2.1" />

  <!-- spawn the left scara robot in gazebo -->
  <node name="spawn_scara_robot_left" pkg="gazebo_ros" type="spawn_model" args="-file $(find two_scara_collaboration)/urdf/scara_robot_left.urdf -urdf -model scara_robot_left
    -y 1.5
    -J scara_robot_left::rotation1 -0.78
    -J scara_robot_left::rotation2 2.1" />


  <!-- the cylinder blocks spawner node -->
  <node name="cylinder_blocks_spawner" pkg="two_scara_collaboration" type="cylinder_blocks_spawner" output="screen" />

  <!-- the cylinder poses publisher node -->
  <node name="cylinder_blocks_poses_publisher" pkg="two_scara_collaboration" type="cylinder_blocks_poses_publisher" output="screen" />

  <!-- the gripper control action server node -->
  <node name="scara_gripper_action_server" pkg="two_scara_collaboration" type="scara_gripper_action_server" output="log" />

  <!-- the scara robot joint controller node -->
  <node name="scara_joint_controller" pkg="two_scara_collaboration" type="scara_joint_controller" output="log" />

  <!-- the cylinder active pool maintainer node -->
  <node name="cylinder_active_pool" pkg="two_scara_collaboration" type="cylinder_active_pool" output="screen" />

  <!-- the maintainer for scara upper boundary topics -->
  <node name="scara_upper_boundary_maintainer" pkg="two_scara_collaboration" type="scara_upper_boundary_maintainer" output="log" />





  <!-- for ros control to be used with scara robot -->
<!--   <param name="/scara_robot_left/robot_description" textfile="$(find two_scara_collaboration)/urdf/scara_robot_left.urdf" /> -->



</launch>

